src/core/FilePath.cpp

```diff
@@ -211,6 +211,9 @@ FilePath::FilePath()
     else if (testSetDir(appDirPath + "/share")) {
     }
 #endif
+    // Last ditch test when running in the build directory (mainly for travis tests)
+    else if (testSetDir(QString(KEEPASSX_SOURCE_DIR) + "/share")) {
```

Comment:
Why adding this duplicate of QT_DEBUG?
On Travis if you are building with the Release flag you need first to run `make install` to install icons and assets and then run tests/guitest. (I've reported this some commit ago but went unnoticed)
This is just a "bad" workaround to make things works.